x-pack/plugins/serverless_search/public/application/components/index_management/index_overview.tsx

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FunctionComponent } from 'react';
+import numeral from '@elastic/numeral';
+import { CoreStart } from '@kbn/core/public';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage, FormattedPlural } from '@kbn/i18n-react';
+import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
+import {
+  EuiLoadingSpinner,
+  EuiEmptyPrompt,
+  EuiFlexGrid,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiIcon,
+  EuiI18nNumber,
+  EuiText,
+  EuiBadge,
+  EuiButton,
+  EuiLink,
+  EuiSpacer,
+} from '@elastic/eui';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { ReactQueryDevtools } from '@tanstack/react-query-devtools';
+
+import { Index } from '@kbn/index-management-plugin/common/types/indices';
+import { IndexContent } from '@kbn/index-management-plugin/public/services';
+
+import { docLinks } from '../../../../common/doc_links';
+import { ServerlessSearchPluginStartDependencies } from '../../../types';
+import { useIndex } from '../../hooks/api/use_index';
+
+import { BadgeList } from '../badge_list';
+import { OverviewEmptyPrompt } from './overview_empty_prompt';
+import { IndexOverviewPanel, IndexOverviewPanelStat } from './overview_panel';
+import { IndexAliasesFlyout } from './index_aliases_flyout';
+
+export interface IndexDetailOverviewProps {
+  index: Index;
+}
+
+export const IndexDetailOverview: FunctionComponent<IndexDetailOverviewProps> = ({ index }) => {
+  const [aliasesFlyoutOpen, setAliasesFlyoutOpen] = React.useState<boolean>(false);
+  const { data, isLoading, isError } = useIndex(index.name);
+  const indexAliases =
+    typeof index.aliases === 'string'
+      ? index.aliases.length > 0 && index.aliases !== 'none'
+        ? [index.aliases]
+        : []
+      : index.aliases;
+
+  if (isLoading || !data)
+    return (
+      <EuiEmptyPrompt
+        icon={<EuiLoadingSpinner size="l" />}
+        title={
+          <h2>
+            <FormattedMessage
+              id="xpack.serverlessSearch.indexManagement.indexDetails.overview.loading.title"
+              defaultMessage="Loading Index"
```

Comment:
nit: there's some inconsistency with capitalizing `Index` in this file. Capitalizing it reads weird to me but I'm not sure what the standard is.